### ISSUE TITLE
`append_domain` can lead to account impersonation

### ISSUE THREAD

**Issue by doobry-systemli** [bit/matrix-synapse-imap#7]
When authenticating against an IMAP server that provides accounts for different domains, `append_domain` can result in account impersonation.

If I login against Matrix with an email address (e.g. `alice@another.org`) different than the Matrix home domain (e.g. `example.org`), the login is successful (due to correct IMAP authentication). But on the Matrix server, I'm authenticated as `@alice:example.org`.

In other words, all IMAP accounts with localpart `alice` on my IMAP server can authenticate as matrix account `@alice:<matrix home domain>`.

We stopped using `append_domain` after discovering this, as it obviously is a major security issue.

The easiest fix probably would be to override the domain part of usernames with the `append_domain` value if set.

**Comment by bit** [comment]
you also had plain_uid set to True?
